Midtrans/ApiRequestor.php

```diff
@@ -111,9 +125,18 @@ public static function remoteCall($url, $server_key, $data_hash, $post = true)
             $curl_options = array_replace_recursive($curl_options, Config::$curlOptions, $headerOptions);
         }
 
-        if ($post) {
+        if ($method == 'POST') {
             $curl_options[CURLOPT_POST] = 1;
 
+            if ($data_hash) {
```

Comment:
This code looks redundant, I think will be better to simplify the code. 
Check the value from method variable if not GET method, then run this line
```
if ($data_hash) {
    $body = json_encode($data_hash);
    $curl_options[CURLOPT_POSTFIELDS] = $body;
} else {
     $curl_options[CURLOPT_POSTFIELDS] = '';
}
```

Reply:
looks good